webapp/src/components/NewsList.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div>
+    <news-item
+        v-for="newsItem in sortedItems"
+        v-bind:title="newsItem.title"
+        v-bind:votes="newsItem.votes"
+        v-bind:key="newsItem.title"
+        v-on:upvoteItem="upvoteNews"
+        v-on:downvoteItem="downvoteNews"
+        v-on:removeNews="removeNews"
```

Comment:
```suggestion
        @updateNews="updateNews"
        @removeNews="removeNews"
```
You should use the same event for updating an item. The parent component shouldn't know any internals of the item, it's enough to know which item has changed and replace this very item. That's it.